File: processweb/src/main/java/com/pulmwine/processweb/command/impl/BaseLevelCommands.java
package com.pulmwine.processweb.command.impl;

import android.content.Context;
import com.alibaba.android.arouter.launcher.ARouter;
import com.pulmwine.processweb.bridge.call.ResultCallback;
import com.pulmwine.processweb.command.Command;
import com.pulmwine.processweb.command.Commands;
import com.pulmwine.processweb.common.WebConstants;

import java.util.Map;

/**
 * Created by xud on 2017/12/16.
 */
public class BaseLevelCommands extends Commands {

    public BaseLevelCommands() {
        registerCommands();
    }

    @Override
    public int getCommandLevel() {
        return WebConstants.LEVEL_BASE;
    }

    private void registerCommands() {
        registerCommand(pageRouterCommand);
    }

    /**
     * 页面路由
     */
    private final Command pageRouterCommand = new Command() {

        @Override
        public String name() {
            return "router";
        }

        @Override
        public void exec(Context context, Map params, ResultCallback resultBack) {
            String newUrl = params.get("url").toString();
            ARouter.getInstance().build(newUrl).navigation();
        }
    };
}

File: CommonWebView/src/main/java/com/plumcookingwine/webview/command/BaseLevelCommands.kt
package com.plumcookingwine.webview.command

import android.content.Context
import com.plumcookingwine.webview.callback.ResultBack
import com.plumcookingwine.webview.command.manager.Command
import com.plumcookingwine.webview.command.manager.Commands
import com.plumcookingwine.webview.common.WebConstants

/**
 * Created by xud on 2017/12/16.
 */

class BaseLevelCommands : Commands() {

    /**
     * 页面路由
     */
    private val pageRouterCommand = object : Command {
        override fun name(): String {
            return "newPage"
        }

        override fun exec(context: Context, params: Map<String, String>, resultBack: ResultBack) {
            val newUrl = params["url"]!!.toString()
            val title = params["title"] as String?
        }
    }

    init {
        registerCommands()
    }

    override fun getCommandLevel(): Int {
        return WebConstants.LEVEL_BASE
    }

    internal fun registerCommands() {
        registerCommand(pageRouterCommand)
    }
}
